src/datasources/postgres/InstrumentDataSource.ts

```diff
@@ -7,33 +12,64 @@ export default class PostgresInstrumentDataSource
 {
   private TABLE_NAME = 'instrument';
 
-  async create(instrument: Instrument): Promise<Instrument> {
-    await database(this.TABLE_NAME).insert({
-      id: instrument.id,
-      name: instrument.name,
-    });
-
-    return instrument;
+  async create(
+    instrument: InstrumentCreationEventPayload
+  ): Promise<Instrument> {
+    return await database(this.TABLE_NAME)
+      .insert({
+        id: instrument.id,
+        name: instrument.name,
+      })
+      .returning(['*'])
+      .then((instrument: InstrumentRecord[]) => {
+        return createInstrumentObject(instrument[0]);
+      });
   }
 
-  async update(instrument: Instrument): Promise<Instrument> {
-    await database(this.TABLE_NAME)
+  async update(
+    instrument: InstrumentUpdationEventPayload
+  ): Promise<Instrument> {
+    const instrumentExists = await database(this.TABLE_NAME)
       .where({
         id: instrument.id,
       })
-      .update({
+      .first();
+
+    if (instrumentExists) {
```

Comment:
note: I think you should not change this, but adding a small comment for knowledge sharing. I believe there is a proper upsert with Postgres and knex which you could use elsewhere which goes something like this
```
const updatedInstrument = await database(this.TABLE_NAME)
    .insert({
      id: instrument.id,
      name: instrument.name,
    })
    .onConflict('id')
    .merge({
      name: instrument.name,
    })
    .returning(['*']);
```